### ISSUE TITLE
[harfbuzz] build failure

### ISSUE THREAD

**Issue by aerworker** [microsoft/vcpkg#10075]
[install-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/4205960/install-x86-windows-dbg-out.log)
**Host Environment**
- OS: Windows 7
- Compiler: 19.00.24215.1 for x86

**To Reproduce**
`vcpkg.exe install harfbuzz[icu] --triplet x86-windows --recurse`

**Failure logs** 
```
Computing installation plan...
The following packages will be built and installed:
    harfbuzz[core,icu,ucdn]:x86-windows
Starting package 1/1: harfbuzz:x86-windows
Building package harfbuzz[core,icu,ucdn]:x86-windows...
Warning: abi keys are missing values:
    freetype
    ragel

-- Downloading https://github.com/harfbuzz/harfbuzz/archive/2.5.3.tar.gz...
-- Extracting source C:/work/vcpkg/downloads/harfbuzz-harfbuzz-2.5.3.tar.gz
-- Applying patch 0001-fix-cmake-export.patch
-- Applying patch 0002-fix-uwp-build.patch
-- Applying patch 0003-remove-broken-test.patch
-- Applying patch find-package-freetype-2.patch
-- Applying patch glib-cmake.patch
-- Using source at C:/work/vcpkg/buildtrees/harfbuzz/src/2.5.3-984271955b
-- Configuring x86-windows
-- Building x86-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:136 (message):
    Command failed: C:/work/vcpkg/downloads/tools/cmake-3.14.0-windows/cmake-3.14.0-win32-x86/bin/cmake.exe --build . --config Debug --target install -- -v
    Working Directory: C:/work/vcpkg/buildtrees/harfbuzz/x86-windows-dbg
    See logs for more information:
      C:\libgdiplus\libgdiplus_cario\vcpkg\buildtrees\harfbuzz\install-x86-windows-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:91 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/harfbuzz/portfile.cmake:72 (vcpkg_install_cmake)
  scripts/ports.cmake:90 (include)
```
[config-x86-windows-out.log](https://github.com/microsoft/vcpkg/files/4205944/config-x86-windows-out.log)
[install-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/4205971/install-x86-windows-dbg-out.log)

**Comment by timautry** [comment]
See https://github.com/harfbuzz/harfbuzz/issues/1990 and https://github.com/harfbuzz/harfbuzz/issues/1993

It is an issue with a missing semi-colon on line 53 of hb-icu.cc 
- #define HB_ICU_STMT(S) do { S } while (0) 
 should read
- #define HB_ICU_STMT(S) do { S; } while (0)

Later versions of harfbuzz have this issue resolved as vcpkg is using v2.5.3 and the github/harfbuzz latest is v.2.6.4

**Comment by PhoebeHui** [comment]
@timautry, thanks for the helpful information! I will have a look at this issue.
